dali/core/mm/default_resources.cc

```diff
@@ -127,24 +128,110 @@ CallAtExit<Callable> AtExit(Callable &&c) {
   return CallAtExit<Callable>(std::forward<Callable>(c));
 }
 
+bool UseDeviceMemoryPool() {
+  static bool value = []() {
+    const char *env = std::getenv("DALI_USE_DEVICE_MEM_POOL");
+    return env && atoi(env);
+  }();
+  return value;
+}
+
+bool UsePinnedMemoryPool() {
+  static bool value = []() {
+    const char *env = std::getenv("DALI_USE_PINNED_MEM_POOL");
+    return !env || atoi(env);
+  }();
+  return value;
+}
+
+bool UseVMM() {
+  static bool value = []() {
+    const char *env = std::getenv("DALI_USE_VMM");
+    return !env || atoi(env);
+  }();
+  return value;
+}
+
+bool UseDeferredDealloc() {
+  static bool value = []() {
+    const char *env = std::getenv("DALI_USE_DEFERRED_DEALLOC");
+    return !env || atoi(env);
+  }();
+  return value;
+}
+
 inline std::shared_ptr<device_async_resource> CreateDefaultDeviceResource() {
   static CUDARTLoader CUDAInit;
 #ifdef DALI_USE_RMM_DEVICE_RESOURCE
   auto upstream = std::make_shared<rmm::mr::cuda_memory_resource>();
   return rmm::mr::make_owning_wrapper<rmm::mr::pool_memory_resource>(std::move(upstream));
 #else
-  using resource_type = mm::async_pool_resource<mm::memory_kind::device>;
-  static auto upstream = std::make_shared<mm::cuda_malloc_memory_resource>();
-  auto rsrc = std::make_shared<resource_type>(upstream.get());
-  return make_shared_composite_resource(std::move(rsrc), upstream);
+  if (!UseDeviceMemoryPool()) {
```

Comment:
Shouldn't the code in 166-167 be in the else for this or something? Here we are creating non-pooled rmm::mr::cuda_memory_resource, and in the else block of `DALI_USE_RMM_DEVICE_RESOURCE` we create non-pooled rmm resource?

Reply:
I guess they should.